### ISSUE TITLE
sChain throws non-behavior exception while couldn't connect to node

### ISSUE THREAD

**Issue by cstrangedk** [skalenetwork/skaled#408]
After turn off sChain container on one of 16 node, in sChain log (on working machine) `Exception: There is no such key in cache` appears, while trying to connect to stopping node:

```java
[2020-11-17 16:40:47.110] [26:main] [error] 28:!Exception: There is no such key in cache
[2020-11-17 16:40:50.111] [26:main] [error] 28:!Exception: ClientSocket:Couldnt connect to:18.193.89.240:10011:Connection refused
[2020-11-17 16:40:50.445] [26:consensus] [info] 28:BLOCK_DECIDE: PRPSR:3:BID:29:STATS:|1|D0R2P46L67|2|D0R2P29L80|3|D1R0P21L87|4|D1R0P18L89|5|D1R0P21L87|6|D1R0P24L87|7|*|8|D1R0P29L84|9|D1R0P43L83|10|D1R0P26L82|11|D1R0P18L80|12|D0R2P99L79|13|*|14|*|15|D1R0P4L74|16|*|| Now signing block ...
[2020-11-17 16:40:50.523] [26:consensus] [info] 28:BLOCK_DECIDE (GOT SIG): PRPSR:3:BID:29| Now signing block ...
[2020-11-17 16:40:50.523] [26:main] [info] 28:BLOCK_SIGNED: Now finalizing block ... BID:29
[2020-11-17 16:40:50.523] [26:main] [info] 28:BLOCK_COMMIT: PRPSR:3:BID: 29:ROOT:46320509353513273106582423493727320152202237096314791991810382902766530930981:HASH:6ea6990c:BLOCK_TXS:0:DMSG:22:MPRPS:5:RPRPS:45:TXS:0:TXLS:50:KNWN:0:MGS:185:INSTS:265:BPS:0:HDRS:94:SOCK:0:CONS:0:DSDS:62
2020-11-17 16:40:50.524656   createBlock ID = #29
```

Note: 18.193.89.240:10011 - stopped node. Test log below

STR:

Run  recovery test  from skale-test-cli

┆Attachments: <a href="https://skalelabs.atlassian.net/secure/attachment/10357/noisy-gorgonea-secunda.zip">noisy-gorgonea-secunda.zip</a>

**Comment by cstrangedk** [comment]
➤ Stan Kladko commented:

The exception comes from lru_cache datastructure . Need to make all calls to lru_cache thread-safe

**Comment by cstrangedk** [comment]
➤ Ganna Kulikova commented:

Not critical for the release

**Comment by cstrangedk** [comment]
Fixed by #421
